presto-main/src/main/java/com/facebook/presto/server/StatementResource.java

```diff
@@ -411,7 +411,7 @@ else if (queryInfo.getOutputStage() == null) {
             // first time through, self is null
             QueryResults queryResults = new QueryResults(
                     queryId.toString(),
-                    uriInfo.getRequestUriBuilder().replaceQuery("").replacePath(queryInfo.getSelf().getPath()).build(),
+                    uriInfo.getRequestUriBuilder().replaceQuery(queryId.toString()).replacePath("query.html").build(),
```

Comment:
I think we should add a new field for this as some people maybe expecting this uri to be the full query info.